### ISSUE TITLE
New Module

### ISSUE THREAD

**Issue by sidhu1012** [sympy/sympy#20017]
Physics module shall have Electrostatics as its part.

**Comment by sidhu1012** [comment]
which module contains already defined sympify symbols like pi ?

**Comment by sylee957** [comment]
`pi` is contained in `from sympy import S`

**Comment by oscarbenjamin** [comment]
> Physics module shall have Electrostatics as its part.

Can you explain what you mean?

What would an electrostatics module in sympy do? Who would use it and for what?

**Comment by sidhu1012** [comment]
> > Physics module shall have Electrostatics as its part.
> 
> Can you explain what you mean?
> 
> What would an electrostatics module in sympy do? Who would use it and for what?

I mean physics module shall have a electrostatics module which should further be divided into particle and body.

Particle would work for point charges and would give value for electrostatic force between 2 charged particles , electric potential due to point charges , dipole moment , electric flux, electric flux etc

Body would contain charged bodies like charged sheet, charged sphere , charged hollow sphere etc and their applications like electric field intensity due to them, charge transfer etc

If we use vectors with electrostatics  we can even add magnetics to it as charge on moving or when places in electric field produces magnetic effect along electric effect.

By adding this I think sympy  would become more powerful tool and would cover wider area in physics.

**Comment by sidhu1012** [comment]
I have been working on it since I raised this issue  and have developed a basic electrostatics module for point charge which is almost done, that PR would give a slightly better idea of what I mean and what my vision is.

**Comment by oscarbenjamin** [comment]
I don't think that we should begin to construct modules like this without a high-level design specified somewhere.

**Comment by sidhu1012** [comment]
> I don't think that we should begin to construct modules like this without a high-level design specified somewhere.

Can you please explain

**Comment by oscarbenjamin** [comment]
If you want to start working on an electrostatics module and open a PR to collect feedback on the PR then that's great.

However I don't think we should merge a PR introducing a new electrostatics module until there is a document somewhere that specifies the intended scope of the module.

What would the module look like when it is completed?

Who would use the module and for what kind of problems?

How would the module relate to other parts of sympy e.g. mechanics?

Many of the concepts you referred to like dipole moment, flux etc are things that can be defined mathematically without reference to electrostatics and that can be useful in other areas of maths and physics. It would probably make sense to add those as part of sympy generally rather than as part of an electrostatics module. Then an electrostatics module could make use of them and they can also be used by other parts of sympy.

We need a plan that is sufficiently detailed that other contributors will be able to keep working on this even after the creator of the module has moved on. There are various modules in sympy that were created by someone who no longer contributes to sympy so that they have become unmaintained. Other contributors would like to improve those modules but it's not clear any more what the plan ever was because no one wrote down any design anywhere.

**Comment by thambi999** [comment]
Not understand ,I not got English ok please help on this

Sent from Yahoo Mail on Android 
 
  On Tue, Sep 1, 2020 at 2:24, Oscar Benjamin<notifications@github.com> wrote:   


If you want to start working on an electrostatics module and open a PR to collect feedback on the PR then that's great.

However I don't think we should merge a PR introducing a new electrostatics module until there is a document somewhere that specifies the intended scope of the module.

What would the module look like when it is completed?

Who would use the module and for what kind of problems?

How would the module relate to other parts of sympy e.g. mechanics?

Many of the concepts you referred to like dipole moment, flux etc are things that can be defined mathematically without reference to electrostatics and that can be useful in other areas of maths and physics. It would probably make sense to add those as part of sympy generally rather than as part of an electrostatics module. Then an electrostatics module could make use of them and they can also be used by other parts of sympy.

We need a plan that is sufficiently detailed that other contributors will be able to keep working on this even after the creator of the module has moved on. There are various modules in sympy that were created by someone who no longer contributes to sympy so that they have become unmaintained. Other contributors would like to improve those modules but it's not clear any more what the plan ever was because no one wrote down any design anywhere.

**Comment by sidhu1012** [comment]
> If you want to start working on an electrostatics module and open a PR to collect feedback on the PR then that's great.
> 
> However I don't think we should merge a PR introducing a new electrostatics module until there is a document somewhere that specifies the intended scope of the module.
> 
> What would the module look like when it is completed?
> 
> Who would use the module and for what kind of problems?
> 
> How would the module relate to other parts of sympy e.g. mechanics?
> 
> Many of the concepts you referred to like dipole moment, flux etc are things that can be defined mathematically without reference to electrostatics and that can be useful in other areas of maths and physics. It would probably make sense to add those as part of sympy generally rather than as part of an electrostatics module. Then an electrostatics module could make use of them and they can also be used by other parts of sympy.
> 
> We need a plan that is sufficiently detailed that other contributors will be able to keep working on this even after the creator of the module has moved on. There are various modules in sympy that were created by someone who no longer contributes to sympy so that they have become unmaintained. Other contributors would like to improve those modules but it's not clear any more what the plan ever was because no one wrote down any design anywhere.

I will be sending a PR to get reviews and get things started. I think module would relate to mechanics quite a bit. Electrostatics is  a wide field . I think we should have a separate module for it.

**Comment by agrawalshubham01** [comment]
I am beginner, but I am interested to work on this module with @sidhu1012 .

**Comment by sidhu1012** [comment]
Does sympy have epsilon naught as a predefined special symbol?

**Comment by sylee957** [comment]
You should take a look at https://github.com/sympy/sympy/blob/e975b8a7b339ee2e209f6d92e8ac58ee278ad5d9/sympy/physics/units/definitions/unit_definitions.py#L274

**Comment by sidhu1012** [comment]
Electrostatics:-

Will use refernce frame and field functions

1.	Class Point_Charge

•	__init__(name, point, charge)
•	Motion of particle under (Electric and Magnetic field, constant and time varying)
•	Electro static field
•	Electro static potential
•	Charge Denisty
•	Electric potential energy
•	Lorentz Force

2.	Class Dipole

•	__init__(name, point_charge_1, point_2, disp=None)
•	Dipole Placed under Electric Field and Magnetic Field(Torque, work etc)
•	 Electric Potential energy

Electro Dynamics(quantum mechanics):-

After this Interaction of Charged Particles with Electromagnetic Radiation can be added and would use Maxwell’ Equations, differential equation.

Example:- 
•	Hamilton Operator
•	Gauge Transformation
•	Schrodinger Equation etc

**Comment by sidhu1012** [comment]
#2361 Have some work in similar Fashion.

**Comment by sidhu1012** [comment]
flux can be added to sympy/physics/vector/fieldfunctions.py
